helm-chart/templates/deployment.yaml

```diff
@@ -8,6 +8,8 @@ spec:
   {{- if not .Values.autoscaling.enabled }}
   replicas: {{ .Values.replicaCount }}
   {{- end }}
+  strategy:
+    type: Recreate
```

Comment:
Doesn't the default RollingUpdate work out?

Reply:
Because the new container can't attach the volume if the old container still alive

Reply:
Wait, that sounds like an implementation problem again. How are other projects doing this if they have volumes mounted to the container?

Reply:
with 
```
spec:
  accessModes:
    - ReadWriteOnce
```
Only one container could mount the volume with that config.
If you want rollingUpdate you must have ReadWriteMany PVC

Reply:
Aha okay. What would be a reasonable/working access mode according to you?

Reply:
We can let this choice to the user, according the storage option available on his infrastructure.
With : 
```
spec:
  accessModes:
    - {{ .Values.pvc.accessModes }} 
 ```
And strategy type according the access modes value

Reply:
Yes, that would be fine for me.

Reply:
Pushed, made a if depends on access mode to define stratigy type : 
``` 
 {{-  if eq .Values.pvc.accessMode "ReadWriteOnce" }}
 strategy:
 type: Recreate
 {{- end }} 
```
If not `ReadWriteOne` use default `strategy.type` > `RollingUpdate`

Reply:
Thanks, there only seems to be a problem with pvc.yaml. It looks like the complete file has been recreated? (Different line endings maybe?)

Reply:
Yes, the file as been created with windows machine and many end line caractere was present, using dos2unix command. 

The previous access mode was not correct regarding the default strategy type, it was `RWO` and `RollingUpdate`, `RWO` must have `Recreate` type

Reply:
Can you fix the line endings for this file? Another PR can be done to change all file's line endings if needed.

I understand, but the added `if` takes care of that, right?

Reply:
> Can you fix the line endings for this file? Another PR can be done to change all file's line endings if needed.

Not sur to understand well. 
The pvc.yaml have been fixed, you want me to do same on all other file ? 

> I understand, but the added if takes care of that, right?

Take care of what ? Above end line caratere ?

Reply:
2 different things:

1. I want you to revert the pvc.yaml line endings change.
2. The block below takes care of the accessMode RWO, so IMO that can stay the default value, right?
```yaml
{{-  if eq .Values.pvc.accessMode "ReadWriteOnce" }}
 strategy:
 type: Recreate
 {{- end }} 
```

If you don't agree with RWO as default, then we can see if we just hard-code it to RWM and remove the strategy section alltogether.

Reply:
Ok I'll revert the pvc. 

Fix an AccessMode is not a good idea, because access mode depend on the storage available on the infrastructure. And ReadWriteMany may not be available. 

In the modification I did, RWM is default (but RWO can also be default as you prefere) is in values.yaml

Reply:
If RWM might not be available, then we definitely should put RWO as default!

Reply:
Right, just wanted to put back your strategy type